gkrellm-plugin.eclass users: Bump to EAPI 8

laumann opened the pull request:
@thesamesam Here's a big PR to track all the users of gkrellm-plugin.eclass, and bumping them to EAPI=8. I'll tick them off as I go, and maybe add comments as well.

 - [x] x11-plugins/gkrellm-cpupower-0.2-r1 - Add patch for Makefile, heavily inspired by the patch file for [x11-plugins/gkrellshoot](https://github.com/gentoo/gentoo/blob/master/x11-plugins/gkrellshoot/files/makefile-respect-flags.patch) by @Flowdalic 
 - [x] x11-plugins/gkrellshoot-0.4.4-r3 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellm-countdown-0.1.2-r1 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellmlaunch-0.5 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellm-mailwatch-2.4.3-r2 -  plus PKG_CONFIG
 - [x] x11-plugins/gkrellm-trayicons-1.03-r1 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellm-vaiobright-2.5-r2
 - [x] x11-plugins/gkrellkam-2.0.0 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellm-xkb-1.05-r1 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellmoon-0.6-r2 - converted `sed` line to patch, plus PKG_CONFIG
 - [ ] x11-plugins/gkrellfire-0.4.2-r1
 - [x] x11-plugins/gkrellsun-1.0.0-r4 - fixes: Don't call `gcc` directly, respect pkgconfig, ensure $(CC) and $(CFLAGS) are used
 - [x] x11-plugins/gkrellweather-2.0.8-r1 - fixes: Don't call `gcc` directly, respect pkgconfig. Suspect that flag-o-matic is not needed
 - [x] x11-plugins/gkrellm-bgchanger-0.1.11-r2
 - [x] x11-plugins/gkrellm-radio-2.0.4 - plus PKG_CONFIG
 - [x] x11-plugins/gkrelltop-2.2.13-r2 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellm-imonc-0.2-r1 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellstock-0.5.1-r1 - plus PKG_CONFIG, rm "-O2 -Wall" flags, don't call gcc directly
 - [x] x11-plugins/gkrellm-volume-2.1.13-r2 - plus PKG_CONFIG
 - [x] x11-plugins/gkrellmss-2.6-r4 - plus PKG_CONFIG, don't call cc directly, respect CFLAGS
 - [x] x11-plugins/i8krellm-2.5-r2 - plus PKG_CONFIG, don't call gcc directly, respect CFLAGS, remove makefile support for gkrellm1
 - [x] x11-plugins/gkrellmwireless-2.0.3-r2 - plus PKG_CONFIG
 - [ ] x11-plugins/gkrellm-bluez-0.2-r2
 - [ ] x11-plugins/gkrellaclock-0.3.4-r1 - bug [742836 (does not respect CFLAGS)](https://bugs.gentoo.org/742836). The compiled .so file fails to load with "gkrellm -p" - it expects a function: `Error: /var/tmp/portage/x11-plugins/gkrellaclock-0.3.4-r2/image/usr/lib64/gkrellm2/plugins/gkrellaclock.so: cannot open shared object file: No such file or directory`



--- 
Overall changes
 - Adjusting Makefiles where appropriate to use user-defined `PKG_CONFIG`

thesamesam commented on x11-plugins/gkrellm-cpupower/files/gkrellm-cpupower-0.2-makefile.patch:
```diff
@@ -0,0 +1,35 @@
+--- a/Makefile
++++ b/Makefile
+@@ -3,17 +3,22 @@
+ GTK_INCLUDE = `pkg-config gtk+-2.0 --cflags`
```
While at it: replace `pkg-config` with `$(PKG_CONFIG)` or similar (defaulting to `pkg-config`) to respect user's choice of pkg-config.

While `pkg-config` should respect appropriate environment variables such that it's not needed to override it really, that doesn't mean much.

laumann replied:
I replaced it with:

```makefile
PKG_CONFIG = pkg-config # is this fine for the default?

GTK_INCLUDE = $(shell ${PKG_CONFIG} gtk+-2.0 --cflags)
GTK_LIB = $(shell ${PKG_CONFIG} gtk+-2.0 --libs)
```

laumann replied:
@thesamesam Should it be doing `tc-export PKG_CONFIG` as in the ebuild as well?

thesamesam replied:
yes!

laumann replied:
OK, then I will probably do:

```
BDEPEND="virtual/pkgconfig"
```
coupled with

```
tc-export PKG_CONFIG
```

in the ebuilds where this patch is added.

laumann replied:
I've gone through the already bumped ebuilds and added patches + BDEPEND for pkg-config.

thesamesam commented on x11-plugins/gkrellm-cpupower/files/gkrellm-cpupower-0.2-makefile.patch:
```diff
@@ -0,0 +1,35 @@
++	$(CC) $(CFLAGS) $(LDFLAGS) $(FLAGS) $(OBJS) -o cpupower.so $(LFLAGS) $(LIBS)
+ 
+ install: cpupower.so
+ 	install -D -m 755 -s cpupower.so $(DESTDIR)/usr/lib/gkrellm2/plugins/cpupower.so
```
It seems odd to me that it's always `lib` but I might be missing something. Also, doubt `-s` should be here (strip).

laumann replied:
I'm not sure about the destination here, but I've gone ahead and removed `-s`.

laumann replied:
I'll remember to check that for the other Makefiles.

thesamesam commented on x11-plugins/gkrellm-cpupower/gkrellm-cpupower-0.2-r2.ebuild:
```diff
@@ -0,0 +1,47 @@
+	"${FILESDIR}"/${P}-makefile.patch
+)
+
+src_install() {
```
Change to match [phase execution order](https://devmanual.gentoo.org/ebuild-writing/functions/index.html)

laumann replied:
Done. Silly me.

thesamesam commented on x11-plugins/gkrellm-cpupower/gkrellm-cpupower-0.2-r2.ebuild:
```diff
@@ -0,0 +1,47 @@
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
[Drop, as it's unnecessary?](Drop, as it's unnecessary?)

laumann replied:
Done.

thesamesam commented on x11-plugins/gkrellshoot/gkrellshoot-0.4.4-r4.ebuild:
```diff
@@ -0,0 +1,26 @@
+	${DEPEND}
+	virtual/imagemagick-tools"
+
+S="${WORKDIR}/${P/s/S}"
```
I prefer S directly below SRC_URI b/c directly related

laumann replied:
OK, I was wondering where best to put - I had gathered between the *DEPEND variables and phase functions :-)

thesamesam commented on x11-plugins/gkrellshoot/gkrellshoot-0.4.4-r4.ebuild:
```diff
@@ -0,0 +1,26 @@
+LICENSE="GPL-2"
+SLOT="2"
+KEYWORDS="~alpha ~amd64 ~ppc ~sparc ~x86"
+IUSE=""
```
Drop

laumann replied:
Done.

laumann commented on x11-plugins/gkrellm-cpupower/files/gkrellm-cpupower-0.2-makefile.patch:
```diff
@@ -0,0 +1,45 @@
+ 
+-GTK_INCLUDE = `pkg-config gtk+-2.0 --cflags`
+-GTK_LIB = `pkg-config gtk+-2.0 --libs`
++PKG_CONFIG = pkg-config
+ 
+-FLAGS = -O2 -Wall -fPIC $(GTK_INCLUDE)
++GTK_INCLUDE = $(shell ${PKG_CONFIG} gtk+-2.0 --cflags)
++GTK_LIB = $(shell ${PKG_CONFIG} gtk+-2.0 --libs)
```
@thesamesam This is what I figured was the "correct" way.

thesamesam replied:
PKG_CONFIG ?= ..., no?

laumann replied:
Yes :-) Changed.

laumann commented:
@juippis made be aware of the fact that ebuilds shouldn't depend on transitive dependencies, and noticed in #24724 that two dependencies were missing from RDEPEND. I think this applies across most (if not all) of these ebuilds. I'll be sure to circle back and add these in where appropriate.

laumann commented:
> 1) I added gkrellfire.o to the end of line 50 of the fixed-fix-CC-CFLAGS-LDFLAGS-handling.patch. 2) I increased CHART_MAX_WIDTH from 120 to 150 in gkrellfire.c. 3) I changed the ebuild SRC_URI to "[http://distcache.FreeBSD.org/ports-distfiles/${P}.tgz](http://distcache.FreeBSD.org/ports-distfiles/$%7BP%7D.tgz)". The old upstream seems dead (as of 0.3.3?,) but I can't figure out where freebsd is getting their source.

1. This is a fix for bug [742839](https://bugs.gentoo.org/742839)? You can just open a separate PR for gkrellfire, then I'll either omit it from this one, or rebase to include your patch
2. The patch for CHART_MAX_WIDTH sounds like a [user patch](https://wiki.gentoo.org/wiki//etc/portage/patches)
3. Is it the same code as from the original SRC_URI?

genterminl commented:
1) not directly for bug 742839.  Not sure why/when I did this, but probably a local build failure.  I'll file separately if it's still needed after your changes.
2) agreed as user patch, but might be useful to avoid surprise if user widens the gkrellm display and grkrellfire doesn't expand to the full width.  Certainly not critical.
3) the updated SRC_URI is because freebsd moved the download source.  The only file at the old location is a README telling of the new location.
